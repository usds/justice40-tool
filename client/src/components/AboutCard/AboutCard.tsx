import React from 'react';
import {Grid} from '@trussworks/react-uswds';

import LinkTypeWrapper from '../LinkTypeWrapper';
import {hyphenizeString} from '../../../cypress/integration/common/helpers';

// the "body" section is the child object to allow for html versus just text
interface AboutCardProps {
  imgSrc?: string;
  header: string;
  size: 'small' | 'large';
  linkText?: string | JSX.Element;
  url?: string;
  openUrlNewTab?: boolean;
  className?: string;
  internal?:boolean;
  // body:string | React.ReactNode;
}

const AboutCard = (props: React.PropsWithChildren<AboutCardProps>) => {
  if (props.size === 'large') {
    // large are the cards on top
    // note it uses a top className='j40-aboutcard-lg-card'
    return (
      <Grid tablet={{col: true}} gap={'lg'} className={(props.className || '')}>
        <Grid row className={'j40-aboutcard-lg-card'}>
          <Grid tablet={{col: 3}} className={'j40-aboutpage-image-container'}>
            {props.imgSrc && <img
              className={'j40-aboutcard-image'}
              alt={props.header}
              src={props.imgSrc}/>}
          </Grid>

          <Grid tablet={{col: 9}} data-cy={`${hyphenizeString(props.header)}-block`}>
            <Grid row>
              <h2>{props.header}</h2>
              {props.children}
            </Grid>
          </Grid>

        </Grid>
      </Grid>
    );
  } else {
    // small are the cards on the bottom
    // note it uses a top className='j40-aboutcard-sm-card'
    return (
      <Grid tablet={{col: true}} gap={'lg'} className={(props.className || '')}>
        <Grid row className={'j40-aboutcard-sm-card'}>
          <Grid tablet={{col: 2}} className={'j40-aboutpage-image-container'}>
            <img
              className={'j40-aboutcard-image'}
              alt={props.header}
              src={props.imgSrc}/>
          </Grid>

          <Grid tablet={{col: 9}} data-cy={`${hyphenizeString(props.header)}-block`}>
            <Grid row>
              <h3>{props.header}</h3>
              {props.children}
              <div className={'j40-aboutcard-sm-link'}>
                {props.url && <LinkTypeWrapper
                  linkText={props.linkText}
                  internal={props.internal}
                  url={props.url ? props.url : ''}
                  openUrlNewTab={props.openUrlNewTab}
                  // className={'j40-aboutcard-link'}
                />}
              </div>
            </Grid>
          </Grid>
          <Grid col={1}>{' '}</Grid>
        </Grid>
      </Grid>
    );
  }
};

export default AboutCard;
