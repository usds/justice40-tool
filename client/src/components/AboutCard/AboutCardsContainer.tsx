import React from 'react';
import {Grid} from '@trussworks/react-uswds';

type AboutCardsContainerProps = {
  className?: string
}

// note: this assumes a J40MainGridContainer container!
const AboutCardsContainer = (props: React.PropsWithChildren<AboutCardsContainerProps>) => {
  return (
    <Grid row gap={'lg'} className={'j40-aboutcard-container ' + (props.className || '')}>
      {props.children}
    </Grid>
  );
};

export default AboutCardsContainer;
