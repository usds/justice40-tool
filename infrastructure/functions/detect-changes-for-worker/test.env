REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1