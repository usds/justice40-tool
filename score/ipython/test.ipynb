{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
