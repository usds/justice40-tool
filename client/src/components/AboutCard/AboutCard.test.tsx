import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import AboutCard from './AboutCard';

describe('rendering of the AboutCard', () => {
  it('checks if small cards component renders', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <AboutCard
            imgSrc={'about:blank'}
            header={'Test Header'}
            size={'small'}
            linkText={'Test Action'}
            url={'#'}>
          Content body of the action card.
          </AboutCard>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if large cards component renders', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <AboutCard
            imgSrc={'about:blank'}
            header={'Test Header'}
            size={'large'}
            linkText={'Test Action'}
            url={'#'}>
          Content body of the action card.
          </AboutCard>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
