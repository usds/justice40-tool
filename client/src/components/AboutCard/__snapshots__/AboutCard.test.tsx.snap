// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the AboutCard checks if large cards component renders 1`] = `
<DocumentFragment>
  <div
    class="grid-gap-lg tablet:grid-col"
    data-testid="grid"
  >
    <div
      class="grid-row j40-aboutcard-lg-card"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-3 j40-aboutpage-image-container"
        data-testid="grid"
      >
        <img
          alt="Test Header"
          class="j40-aboutcard-image"
          src="about:blank"
        />
      </div>
      <div
        class="tablet:grid-col-9"
        data-cy="test-header-block"
        data-testid="grid"
      >
        <div
          class="grid-row"
          data-testid="grid"
        >
          <h2>
            Test Header
          </h2>
          Content body of the action card.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering of the AboutCard checks if small cards component renders 1`] = `
<DocumentFragment>
  <div
    class="grid-gap-lg tablet:grid-col"
    data-testid="grid"
  >
    <div
      class="grid-row j40-aboutcard-sm-card"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-2 j40-aboutpage-image-container"
        data-testid="grid"
      >
        <img
          alt="Test Header"
          class="j40-aboutcard-image"
          src="about:blank"
        />
      </div>
      <div
        class="tablet:grid-col-9"
        data-cy="test-header-block"
        data-testid="grid"
      >
        <div
          class="grid-row"
          data-testid="grid"
        >
          <h3>
            Test Header
          </h3>
          Content body of the action card.
          <div
            class="j40-aboutcard-sm-link"
          >
            <a
              data-cy=""
              href="#"
            >
              Test Action
            </a>
          </div>
        </div>
      </div>
      <div
        class="grid-col-1"
        data-testid="grid"
      >
         
      </div>
    </div>
  </div>
</DocumentFragment>
`;
